File: src/lib/finalTouches.ts
import { invoke } from "@tauri-apps/api/core";

export type FinalTouchStepKind = "shell" | "validate" | "commit" | "pull_request";

export type FinalTouchStep = {
  id: string;
  projectPath: string;
  kind: FinalTouchStepKind;
  label: string;
  enabled: boolean;
  sortOrder: number;
  config: Record<string, unknown>;
  createdAt: number;
  updatedAt: number;
};

export type FinalTouchStepInput = {
  projectPath: string;
  kind: string;
  label: string;
  enabled?: boolean;
  sortOrder?: number;
  config?: Record<string, unknown>;
};

export async function listFinalTouchSteps(
  projectPath: string,
): Promise<FinalTouchStep[]> {
  return invoke<FinalTouchStep[]>("final_touch_list_steps", { projectPath });
}

export async function createFinalTouchStep(
  input: FinalTouchStepInput,
): Promise<FinalTouchStep> {
  return invoke<FinalTouchStep>("final_touch_create_step", { input });
}

export async function updateFinalTouchStep(
  id: string,
  input: FinalTouchStepInput,
): Promise<FinalTouchStep> {
  return invoke<FinalTouchStep>("final_touch_update_step", { id, input });
}

export async function setFinalTouchStepEnabled(
  id: string,
  enabled: boolean,
): Promise<void> {
  return invoke<void>("final_touch_set_enabled", { id, enabled });
}

export async function reorderFinalTouchStep(
  id: string,
  newOrder: number,
): Promise<void> {
  return invoke<void>("final_touch_reorder_step", { id, newOrder });
}

export async function deleteFinalTouchStep(id: string): Promise<void> {
  return invoke<void>("final_touch_delete_step", { id });
}

File: src-tauri/src/commands/final_touches.rs
use crate::services::final_touches_service::{FinalTouchStep, FinalTouchStepInput, FinalTouchesService};

#[tauri::command]
pub fn final_touch_list_steps(project_path: String) -> Result<Vec<FinalTouchStep>, String> {
    FinalTouchesService::list_steps(&project_path)
}

#[tauri::command]
pub fn final_touch_create_step(input: FinalTouchStepInput) -> Result<FinalTouchStep, String> {
    FinalTouchesService::create_step(input)
}

#[tauri::command]
pub fn final_touch_update_step(
    id: String,
    input: FinalTouchStepInput,
) -> Result<FinalTouchStep, String> {
    FinalTouchesService::update_step(&id, input)
}

#[tauri::command]
pub fn final_touch_set_enabled(id: String, enabled: bool) -> Result<(), String> {
    FinalTouchesService::set_enabled(&id, enabled)
}

#[tauri::command]
pub fn final_touch_reorder_step(id: String, new_order: i64) -> Result<(), String> {
    FinalTouchesService::reorder_step(&id, new_order)
}

#[tauri::command]
pub fn final_touch_delete_step(id: String) -> Result<(), String> {
    FinalTouchesService::delete_step(&id)
}
